woocommerce/woocommerce-android
# [HACK Week] Enable and add Zendesk code unit test coverage

⚠️ Disclaimer
==========
Since this PR adds a full test suite for the Zendesk implementation, from the 711 lines added in this PR, 555 is unit test code. This PR is not as big as it looks.

Summary
==========
This PR applies the final adjustments to the Zendesk group of classes to allow proper unit test coverage by extracting from the `ZendeskTicketRepository` all operations related to the device information like network, logs, language, and app version to an injected dependency defined as `ZendeskEnvironmentDataSource`. 

It also introduces the full test suite for the `ZendeskTicketRepository`.

How to Test
==========
1. Try to mess with the central features of the `ZendeskTicketRepository` and verify if the unit tests catch them.

Update release notes:
